packages/cli-kit/src/session.ts

```diff
@@ -215,6 +217,9 @@ ${token.json(applications)}
     } catch (error) {
       if (error instanceof InvalidGrantError) {
         newSession = await executeCompleteFlow(applications, fqdn)
+      } else if (error instanceof InvalidRequestError) {
+        await secureStore.remove()
+        throw new AbortError('\nError validating auth session', "We've cleared the current session, please try again")
```

Comment:
What about retrying on behalf of users? As a user of a tool, I'd expect to see no information about the internals of the tool (e.g., not telling them that some internal validation failed), and ideally self-healing such that I don't have to do something that the tool could have done for me.

Reply:
This is what we were talking here: https://github.com/Shopify/cli/pull/979#discussion_r1056262144